### DISCUSSION TITLE
Capture delete action

### DISCUSSION THREAD

**Discussion by JuanMaGMZG** [Blazor-Diagrams/Blazor.Diagrams#353] [General]
I want to capture the event of deleting what I have selected (either link or node), but I do not know if there is already a function for it or I have to do it.

diagram.KeyDown += (e) =>
{
    if (e.Key == "Delete")
    {
        e.preventDefault();
    }
};

I tried the diagram.keydown when I press Delete, but e.preventDefault does not exist.

Do you have any suggestions?

Thx 4 all

**Comment by zHaytam** [comment]
Do you want to cancel a deletion based on some condition?

**Reply by JuanMaGMZG** [reply]
I want to offer you a "confirm" and if you answer no, do not delete the selected item

**Comment by JuanMaGMZG** [comment]
if it serves you, what I did was remove the shortcut to delete and in the KeyDown event, get the selected link or the node if it is a node that is selected

```
var ksb = diagram.GetBehavior<KeyboardShortcutsBehavior>();
ksb.RemoveShortcut("Delete", ctrl: false, shift: false, alt: false);

diagram.KeyDown += async (e) =>
{
    if (e.Key == "Delete")
    {
        //Only for links
        var linkToDelete = diagram.Links.FirstOrDefault(link => link.Selected);
        if (linkToDelete != null)
           await HandleDeleteLink(linkToDelete);
    }
};

```

**Reply by ATEEKGIT** [reply]
if you want to control deletion of nodes etc.  just use the constaint ,  like below

// add constrain in option and attach your method which will be called 
 var options = new BlazorDiagramOptions
            {
                Constraints =
            {
                 ShouldDeleteNode = IsDeleteAllowed 
            } 


// then implement method like this.
    public async ValueTask<bool> IsDeleteAllowed(NodeModel DNode)
    {
        var xnode = DNode as TaskNodeModel;
        if (xnode.NodeStep.step_type == StepType.Start || xnode.NodeStep.step_type == StepType.End)
            return false;
        else
            return true; 

    }



this function control which type of node can be deleted and which can't .
